sklearn/metrics/_pairwise_distances_reduction/_argkmin.pyx.tp

```diff
@@ -197,7 +197,6 @@ cdef class PairwiseDistancesArgKmin{{name_suffix}}(PairwiseDistancesReduction{{n
         self.heaps_r_distances_chunks[thread_num] = &self.argkmin_distances[X_start, 0]
         self.heaps_indices_chunks[thread_num] = &self.argkmin_indices[X_start, 0]
 
-    @final
```

Comment:
As a quick check, can we run a small benchmark to make sure removing the `final` does not lead to a performance regression?